### ISSUE TITLE
Color class is not self-consistent

### ISSUE THREAD

**Issue by Brow71189** [nion-software/nionutils#21]
Try something like `Color.Color('rgba(255, 0, 0, 0.5)').to_rgba_255()`
I would expect this to return `(255, 0, 0, 127)` but instead I get `(127, 255, 0, 0)`.

**Comment by cmeyer** [comment]
I'll take a look at the bug, but the color is just working off the CSS spec:

- https://w3c.github.io/csswg-drafts/css-color/#rgb-functions
- https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/rgba

**Comment by cmeyer** [comment]
I assume you're using a background color with transparency? (I think that's the only place where this would show up...)

**Comment by Brow71189** [comment]
Exactly. And there is doesn't work because you simply get a white background instead.
Following the CSS spec is nice, but I think our own color class should work with our own UI, otherwise it is not that useful.

**Comment by Brow71189** [comment]
Maybe the bug should be in "nionui" then and we need to change the "set_background" method to adhere to the CSS spec?

**Comment by lntutor** [comment]
I’m taking this. I’ll add a regression test for CSS `rgba()` conversion to the `(red, green, blue, alpha)` byte order, make the minimal fix, and run the focused color tests.

**Comment by cmeyer** [comment]
Is this issue showing up in real world use in Swift? I'm apprehensive about changing this without understanding its downstream consequences as this is very low level and can show up in places that make assumptions about its [incorrect] behavior.

@lntutor Would it be possible for you to contact us directly through regular channels (not GitHub) and introduce yourself and how you use Swift?
